qgym/generators/interaction.py

```diff
@@ -29,41 +28,57 @@ class InteractionGenerator(
     def __next__(self) -> NDArray[np.int_]:
         """Make a new interaction circuit.
 
-        The __next__ method of a :class:`InteractionGenerator` should generate a NDArray
-        of shape (len_circuit, 2) with dtype int. Each pair represent he indices of two
-        qubits that have an interaction in the circuit.
+        The ``__next__`` method of a :class:`InteractionGenerator` should generate a
+        :class:`~numpy.ndarray` of shape (len_circuit, 2) with dtype ``int``. Each pair
+        represent the indices of two qubits that have an interaction in the circuit.
+        """
+
+    @abstractmethod
+    def set_state_attributes(self, **kwargs: Any) -> None:
+        """Set attributes that the state can receive.
+
+        This method is called inside the scheduling environment to receive information
+        about the state. The same keywords as for the the init of the
+        :class:`~qgym.envs.routing.RoutingState` are provided.
         """
 
 
 class BasicInteractionGenerator(InteractionGenerator):
     """:class:`BasicInteractionGenerator` is an interaction generation implementation.
 
-    Interactions are completely randomly generated using the ``numpy`` `choice`_ method.
-
-    .. _choice: https://numpy.org/doc/stable/reference/random/generated/
-       numpy.random.choice.html
+    Interactions are completely randomly generated using the :func:`numpy.random.choice`
+    method.
     """
 
     def __init__(
         self,
-        n_qubits: SupportsInt,
         max_length: SupportsInt = 10,
         seed: Generator | SupportsInt | None = None,
     ) -> None:
         """Init of the :class:`BasicInteractionGenerator`.
 
         Args:
-            n_qubits: Number of qubits.
             max_length: Maximum length of the generated interaction circuits. Defaults
                 to 10.
             seed: Seed to use.
         """
-        self.n_qubits = check_int(n_qubits, "n_qubits", l_bound=1)
         self.max_length = check_int(max_length, "max_length", l_bound=1)
         self.rng = parse_seed(seed)
         self.finite = False
+        self.n_qubits: int
         super().__init__()
 
+    def set_state_attributes(self, **kwargs: dict[str, Any]) -> None:
+        """Set the `n_qubits` attribute.
+
+        Args:
+            kwargs: Keyword arguments. Must have the ``"connection_graph"`` key with a
+                :class:`~networkx.Graph` representation of the connection graph.
+        """
```

Comment:
Could it be useful to add some if statement that checks whether the connection_graph is in the **kwargs? And if not, raise an error.